pkg/util/slices/slices.go

```diff
@@ -82,3 +82,17 @@ func CmpNoOrder[E comparable, S ~[]E](a, b S) bool {
 	}
 	return true
 }
+
+// Pick creates a new slice containing only the elements for which keep return true.
+func Pick[E any, S ~[]E](s S, keep func(*E) bool) S {
+	if s == nil {
+		return nil
+	}
+	ret := S{}
```

Comment:
nit:
```suggestion
	var ret S
```
To save one conditional and an allocation if no elements pass `keep`